### PR TITLE
509 feature request cloudbrute tool

### PR DESCRIPTION
**Pull request by gssravanthi31**
This pull request introduces feature cloudbrute tool #509 to the toolkit. Testing for the tool is completed and gives the accurate output.

### CODE REVIEW

File: src/components/CloudBrute/CloudBrute.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import { Button, Stack, TextInput } from "@mantine/core";
+import { useForm } from "@mantine/form";
+import { useCallback, useState, useEffect } from "react";
+import { CommandHelper } from "../../utils/CommandHelper";
+import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { RenderComponent } from "../UserGuide/UserGuide";
+import { SaveOutputToTextFile } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:8]: You can use LoadingOverlayAndCancelButtonPkexec 

```suggestion
import { LoadingOverlayAndCancelButtonPkexec } from "../OverlayAndCancelButton/OverlayAndCancelButton";
```

File: src/components/CloudBrute/CloudBrute.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
+import { Button, Stack, TextInput } from "@mantine/core";
+import { useForm } from "@mantine/form";
+import { useCallback, useState, useEffect } from "react";
+import { CommandHelper } from "../../utils/CommandHelper";
+import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { RenderComponent } from "../UserGuide/UserGuide";
+import { SaveOutputToTextFile } from "../SaveOutputToFile/SaveOutputToTextFile";
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:7]: You can use SaveOutputToTextFile_v2 

```suggestion
import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
```

File: src/components/CloudBrute/CloudBrute.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description}
+                    dependencies={dependencies}
+                ></InstallationModal>
+            )}
+            <form onSubmit={form.onSubmit(onSubmit)}>
+                <Stack>
+                    {LoadingOverlayAndCancelButton(loading, pid)}
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:150]: You should use  LoadingOverlayAndCancelButtonPkexec here.
```suggestion
{LoadingOverlayAndCancelButtonPkexec(loading, pid, handleProcessData, handleProcessTermination)}
```

File: src/components/CloudBrute/CloudBrute.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    const title = "CloudBrute"; // Title of the component.
+    const description =
+        "CloudBrute is a tool for cloud enumeration and infrastructure discovery in various cloud providers."; // Description of the component.
+    const steps =
+        "Step 1: Enter the target domain.\n" +
+        "Step 2: Enter a keyword for URL generation.\n" +
+        "Step 3: Specify the path to the wordlist file.\n" +
+        "Step 4: Click 'Run CloudBrute' to start the scan.\n" +
+        "Step 5: View the Output block below to see the results of the scan.";
+    const sourceLink = ""; // Link to the source code.
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:44]: You forgot to add the sourcelink

```suggestion
    const sourceLink = "https://www.kali.org/tools/cloudbrute/"; // Link to the source code.
```

File: src/components/CloudBrute/CloudBrute.tsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+                    />
+                    <TextInput label="Keyword" required placeholder="e.g., test" {...form.getInputProps("keyword")} />
+                    <TextInput
+                        label="Path to Wordlist"
+                        required
+                        placeholder="/usr/share/dirb/wordlists/common.txt"
+                        {...form.getInputProps("wordlist")}
+                    />
+                    {SaveOutputToTextFile(output)}
+                    <Button type="submit">Run CloudBrute</Button>
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:165]: The button should be Start {title}

```suggestion
                    <Button type="submit">Start {title}</Button>
```

File: src/components/CloudBrute/CloudBrute.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+                setOutput(`Error: ${error.message}`);
+            });
+    };
+
+    /**
+     * Clears the output state.
+     */
+    const clearOutput = useCallback(() => {
+        setOutput("");
+    }, [setOutput]);
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:130]: You missed  setAllowSave, setHasSaved and also the Callback function to handle the completion of a save operation.

```suggestion
    const clearOutput = useCallback(() => {
        setOutput("");
        setAllowSave(false);
        setHasSaved(false);
    }, [setOutput]);

     const handleSaveComplete = useCallback(() => {
        // Indicating that the file has saved which is passed
        // back into SaveOutputToTextFile to inform the user
        setHasSaved(true);
        setAllowSave(false);
    }, []);
```

File: src/components/CloudBrute/CloudBrute.tsx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+                    <TextInput label="Keyword" required placeholder="e.g., test" {...form.getInputProps("keyword")} />
+                    <TextInput
+                        label="Path to Wordlist"
+                        required
+                        placeholder="/usr/share/dirb/wordlists/common.txt"
+                        {...form.getInputProps("wordlist")}
+                    />
+                    {SaveOutputToTextFile(output)}
+                    <Button type="submit">Run CloudBrute</Button>
+                    <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
```
Comment (Oliver2049) [inline src/components/CloudBrute/CloudBrute.tsx:166]: You forgot to add {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
```suggestion
                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
                    <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
```

**Comment by chandelshubham63** [comment]
@michaeljpigott All requested changes have been applied, and the tool has been tested and is working properly. It is ready to be merged. Thanks! ;)

**Comment by michaeljpigott** [comment]
Hi @Oliver2049 do you have time to review this pull request?

**Comment by Oliver2049** [comment]
> Hi @Oliver2049 do you have time to review this pull request?

Hi @michaeljpigott ,sure I will review this.

**Review by Oliver2049 — APPROVED** [review]
Hi @gssravanthi31 
Thank you for your effort. It looks good to me.
Regards
Oliver

**Pull request merged**